From: Lorenzo Stoakes <lorenzo.stoakes@oracle.com>
Subject: fs: convert simple use of generic_file_*_mmap() to .mmap_prepare()
Date: Mon, 16 Jun 2025 20:33:27 +0100

Since commit c84bf6dd2b83 ("mm: introduce new .mmap_prepare() file
callback"), the f_op->mmap() hook has been deprecated in favour of
f_op->mmap_prepare().

We have provided generic .mmap_prepare() equivalents, so update all file
systems that specify these directly in their file_operations structures.

This updates 9p, adfs, affs, bfs, fat, hfs, hfsplus, hostfs, hpfs, jffs2,
jfs, minix, omfs, ramfs and ufs file systems directly.

It updates generic_ro_fops which impacts qnx4, cramfs, befs, squashfs,
frebxfs, qnx6, efs, romfs, erofs and isofs file systems.

There are remaining file systems which use generic hooks in a less direct
way which we address in a subsequent commit.

Link: https://lkml.kernel.org/r/c7dc90e44a9e75e750939ea369290d6e441a18e6.1750099179.git.lorenzo.stoakes@oracle.com
Signed-off-by: Lorenzo Stoakes <lorenzo.stoakes@oracle.com>
Reviewed-by: Viacheslav Dubeyko <Slava.Dubeyko@ibm.com>		[hfs/hfs+]
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 fs/9p/vfs_file.c        |    2 +-
 fs/adfs/file.c          |    2 +-
 fs/affs/file.c          |    2 +-
 fs/bfs/file.c           |    2 +-
 fs/fat/file.c           |    2 +-
 fs/hfs/inode.c          |    2 +-
 fs/hfsplus/inode.c      |    2 +-
 fs/hostfs/hostfs_kern.c |    2 +-
 fs/hpfs/file.c          |    2 +-
 fs/jffs2/file.c         |    2 +-
 fs/jfs/file.c           |    2 +-
 fs/minix/file.c         |    2 +-
 fs/omfs/file.c          |    2 +-
 fs/ramfs/file-mmu.c     |    2 +-
 fs/read_write.c         |    2 +-
 fs/ufs/file.c           |    2 +-
 16 files changed, 16 insertions(+), 16 deletions(-)

--- a/fs/9p/vfs_file.c~fs-convert-simple-use-of-generic_file__mmap-to-mmap_prepare
+++ a/fs/9p/vfs_file.c
@@ -516,7 +516,7 @@ const struct file_operations v9fs_file_o
 	.open = v9fs_file_open,
 	.release = v9fs_dir_release,
 	.lock = v9fs_file_lock,
-	.mmap = generic_file_readonly_mmap,
+	.mmap_prepare = generic_file_readonly_mmap_prepare,
 	.splice_read = v9fs_file_splice_read,
 	.splice_write = iter_file_splice_write,
 	.fsync = v9fs_file_fsync,
--- a/fs/adfs/file.c~fs-convert-simple-use-of-generic_file__mmap-to-mmap_prepare
+++ a/fs/adfs/file.c
@@ -25,7 +25,7 @@
 const struct file_operations adfs_file_operations = {
 	.llseek		= generic_file_llseek,
 	.read_iter	= generic_file_read_iter,
-	.mmap		= generic_file_mmap,
+	.mmap_prepare	= generic_file_mmap_prepare,
 	.fsync		= generic_file_fsync,
 	.write_iter	= generic_file_write_iter,
 	.splice_read	= filemap_splice_read,
--- a/fs/affs/file.c~fs-convert-simple-use-of-generic_file__mmap-to-mmap_prepare
+++ a/fs/affs/file.c
@@ -999,7 +999,7 @@ const struct file_operations affs_file_o
 	.llseek		= generic_file_llseek,
 	.read_iter	= generic_file_read_iter,
 	.write_iter	= generic_file_write_iter,
-	.mmap		= generic_file_mmap,
+	.mmap_prepare	= generic_file_mmap_prepare,
 	.open		= affs_file_open,
 	.release	= affs_file_release,
 	.fsync		= affs_file_fsync,
--- a/fs/bfs/file.c~fs-convert-simple-use-of-generic_file__mmap-to-mmap_prepare
+++ a/fs/bfs/file.c
@@ -27,7 +27,7 @@ const struct file_operations bfs_file_op
 	.llseek 	= generic_file_llseek,
 	.read_iter	= generic_file_read_iter,
 	.write_iter	= generic_file_write_iter,
-	.mmap		= generic_file_mmap,
+	.mmap_prepare	= generic_file_mmap_prepare,
 	.splice_read	= filemap_splice_read,
 };
 
--- a/fs/fat/file.c~fs-convert-simple-use-of-generic_file__mmap-to-mmap_prepare
+++ a/fs/fat/file.c
@@ -204,7 +204,7 @@ const struct file_operations fat_file_op
 	.llseek		= generic_file_llseek,
 	.read_iter	= generic_file_read_iter,
 	.write_iter	= generic_file_write_iter,
-	.mmap		= generic_file_mmap,
+	.mmap_prepare	= generic_file_mmap_prepare,
 	.release	= fat_file_release,
 	.unlocked_ioctl	= fat_generic_ioctl,
 	.compat_ioctl	= compat_ptr_ioctl,
--- a/fs/hfs/inode.c~fs-convert-simple-use-of-generic_file__mmap-to-mmap_prepare
+++ a/fs/hfs/inode.c
@@ -690,7 +690,7 @@ static const struct file_operations hfs_
 	.llseek		= generic_file_llseek,
 	.read_iter	= generic_file_read_iter,
 	.write_iter	= generic_file_write_iter,
-	.mmap		= generic_file_mmap,
+	.mmap_prepare	= generic_file_mmap_prepare,
 	.splice_read	= filemap_splice_read,
 	.fsync		= hfs_file_fsync,
 	.open		= hfs_file_open,
--- a/fs/hfsplus/inode.c~fs-convert-simple-use-of-generic_file__mmap-to-mmap_prepare
+++ a/fs/hfsplus/inode.c
@@ -366,7 +366,7 @@ static const struct file_operations hfsp
 	.llseek		= generic_file_llseek,
 	.read_iter	= generic_file_read_iter,
 	.write_iter	= generic_file_write_iter,
-	.mmap		= generic_file_mmap,
+	.mmap_prepare	= generic_file_mmap_prepare,
 	.splice_read	= filemap_splice_read,
 	.fsync		= hfsplus_file_fsync,
 	.open		= hfsplus_file_open,
--- a/fs/hostfs/hostfs_kern.c~fs-convert-simple-use-of-generic_file__mmap-to-mmap_prepare
+++ a/fs/hostfs/hostfs_kern.c
@@ -382,7 +382,7 @@ static const struct file_operations host
 	.splice_write	= iter_file_splice_write,
 	.read_iter	= generic_file_read_iter,
 	.write_iter	= generic_file_write_iter,
-	.mmap		= generic_file_mmap,
+	.mmap_prepare	= generic_file_mmap_prepare,
 	.open		= hostfs_open,
 	.release	= hostfs_file_release,
 	.fsync		= hostfs_fsync,
--- a/fs/hpfs/file.c~fs-convert-simple-use-of-generic_file__mmap-to-mmap_prepare
+++ a/fs/hpfs/file.c
@@ -255,7 +255,7 @@ const struct file_operations hpfs_file_o
 	.llseek		= generic_file_llseek,
 	.read_iter	= generic_file_read_iter,
 	.write_iter	= generic_file_write_iter,
-	.mmap		= generic_file_mmap,
+	.mmap_prepare	= generic_file_mmap_prepare,
 	.release	= hpfs_file_release,
 	.fsync		= hpfs_file_fsync,
 	.splice_read	= filemap_splice_read,
--- a/fs/jffs2/file.c~fs-convert-simple-use-of-generic_file__mmap-to-mmap_prepare
+++ a/fs/jffs2/file.c
@@ -54,7 +54,7 @@ const struct file_operations jffs2_file_
  	.read_iter =	generic_file_read_iter,
  	.write_iter =	generic_file_write_iter,
 	.unlocked_ioctl=jffs2_ioctl,
-	.mmap =		generic_file_readonly_mmap,
+	.mmap_prepare =	generic_file_readonly_mmap_prepare,
 	.fsync =	jffs2_fsync,
 	.splice_read =	filemap_splice_read,
 	.splice_write = iter_file_splice_write,
--- a/fs/jfs/file.c~fs-convert-simple-use-of-generic_file__mmap-to-mmap_prepare
+++ a/fs/jfs/file.c
@@ -143,7 +143,7 @@ const struct file_operations jfs_file_op
 	.llseek		= generic_file_llseek,
 	.read_iter	= generic_file_read_iter,
 	.write_iter	= generic_file_write_iter,
-	.mmap		= generic_file_mmap,
+	.mmap_prepare	= generic_file_mmap_prepare,
 	.splice_read	= filemap_splice_read,
 	.splice_write	= iter_file_splice_write,
 	.fsync		= jfs_fsync,
--- a/fs/minix/file.c~fs-convert-simple-use-of-generic_file__mmap-to-mmap_prepare
+++ a/fs/minix/file.c
@@ -17,7 +17,7 @@ const struct file_operations minix_file_
 	.llseek		= generic_file_llseek,
 	.read_iter	= generic_file_read_iter,
 	.write_iter	= generic_file_write_iter,
-	.mmap		= generic_file_mmap,
+	.mmap_prepare	= generic_file_mmap_prepare,
 	.fsync		= generic_file_fsync,
 	.splice_read	= filemap_splice_read,
 };
--- a/fs/omfs/file.c~fs-convert-simple-use-of-generic_file__mmap-to-mmap_prepare
+++ a/fs/omfs/file.c
@@ -332,7 +332,7 @@ const struct file_operations omfs_file_o
 	.llseek = generic_file_llseek,
 	.read_iter = generic_file_read_iter,
 	.write_iter = generic_file_write_iter,
-	.mmap = generic_file_mmap,
+	.mmap_prepare = generic_file_mmap_prepare,
 	.fsync = generic_file_fsync,
 	.splice_read = filemap_splice_read,
 };
--- a/fs/ramfs/file-mmu.c~fs-convert-simple-use-of-generic_file__mmap-to-mmap_prepare
+++ a/fs/ramfs/file-mmu.c
@@ -41,7 +41,7 @@ static unsigned long ramfs_mmu_get_unmap
 const struct file_operations ramfs_file_operations = {
 	.read_iter	= generic_file_read_iter,
 	.write_iter	= generic_file_write_iter,
-	.mmap		= generic_file_mmap,
+	.mmap_prepare	= generic_file_mmap_prepare,
 	.fsync		= noop_fsync,
 	.splice_read	= filemap_splice_read,
 	.splice_write	= iter_file_splice_write,
--- a/fs/read_write.c~fs-convert-simple-use-of-generic_file__mmap-to-mmap_prepare
+++ a/fs/read_write.c
@@ -28,7 +28,7 @@
 const struct file_operations generic_ro_fops = {
 	.llseek		= generic_file_llseek,
 	.read_iter	= generic_file_read_iter,
-	.mmap		= generic_file_readonly_mmap,
+	.mmap_prepare	= generic_file_readonly_mmap_prepare,
 	.splice_read	= filemap_splice_read,
 };
 
--- a/fs/ufs/file.c~fs-convert-simple-use-of-generic_file__mmap-to-mmap_prepare
+++ a/fs/ufs/file.c
@@ -38,7 +38,7 @@ const struct file_operations ufs_file_op
 	.llseek		= generic_file_llseek,
 	.read_iter	= generic_file_read_iter,
 	.write_iter	= generic_file_write_iter,
-	.mmap		= generic_file_mmap,
+	.mmap_prepare	= generic_file_mmap_prepare,
 	.open           = generic_file_open,
 	.fsync		= generic_file_fsync,
 	.splice_read	= filemap_splice_read,
_
